uniffi/src/bindings/swift/mod.rs

```diff
@@ -112,68 +106,43 @@ pub fn compile_swift_module(ci: &ComponentInterface, out_dir: &OsStr) -> Result<
         .arg("-Xcc")
         .arg(module_map_file_option)
         .arg(source_file)
-        .spawn()
-        .map_err(|_| anyhow::anyhow!("failed to spawn `swiftc`"))?
-        .wait()
-        .map_err(|_| anyhow::anyhow!("failed to wait for `swiftc` subprocess"))?;
+        .spawn()?
+        .wait()?;
     if !status.success() {
         bail!("running `swiftc` failed")
     }
     Ok(())
 }
 
-pub fn run_swift_script(out_dir: Option<&OsStr>, script_file: Option<&str>) -> Result<()> {
-    // TODO: Don't hard-code library names for the REPL.
-    const LIBS: &[&str] = &["-larithmetic", "-lgeometry", "-lsprites"];
-    const MODULE_MAPS: &[&str] = &[
-        "uniffi_arithmetic.modulemap",
-        "uniffi_geometry.modulemap",
-        "uniffi_sprites.modulemap",
-    ];
-
+pub fn run_script(out_dir: Option<&Path>, script_file: Option<&Path>) -> Result<()> {
     let mut cmd = std::process::Command::new("swift");
-    cmd.args(LIBS);
 
+    // Find any module maps and/or dylibs in the target directory, and tell swift to use them.
     if let Some(out_dir) = out_dir {
-        let out_path = PathBuf::from(out_dir);
-        let cc_options: Vec<OsString> = MODULE_MAPS
-            .iter()
-            .flat_map(|module_map| {
-                let mut module_map_file = out_path.clone();
-                module_map_file.push(module_map);
-                if !module_map_file.exists() {
-                    // Missing module maps (for example, `uniffi_arithmetic` when we're
-                    // running the `geometry` example) will fail, so ignore maps that
-                    // don't exist. Gross!
-                    return Vec::new();
+        cmd.arg("-I").arg(out_dir).arg("-L").arg(out_dir);
+        for entry in PathBuf::from(out_dir).read_dir()? {
```

Comment:
@linacambridge here I've tried to address https://github.com/rfk/uniffi-rs/issues/16 by looking for files in the target directory, instead of hard-coding them. It seems to work in my local testing. WDYT?